### ISSUE TITLE
Failing tests on 5.5.0

### ISSUE THREAD

**Issue by HorstSteuer** [ANR-COMPASS/shesha#60]
On a fresh install (`conda install -c "compass/label/cuda12" compass`) a number of tests fail for me (I guess having a random Strehl value means that the test passed):
```
|    | Test name                      | Init   | SR@100iter          |   T Init |    T Loop |
|---:|:-------------------------------|:-------|:--------------------|---------:|----------:|
|  0 | test_custom_dm_diam_dm_proj.py | False  | N/A                 |        0 | 0         |
|  1 | test_custom_dm_diam_dm.py      | False  | N/A                 |        0 | 0         |
|  2 | test_custom_dm_pitch.py        | False  | N/A                 |        0 | 0         |
|  3 | test_custom_dm.py              | False  | N/A                 |        0 | 0         |
|  4 | test_pyr3_maskedpix.py         | True   | 0.7534236907958984  |        0 | 0.751669  |
|  5 | test_pyr_base.py               | True   | 0.6425969004631042  |        0 | 0.721587  |
|  6 | test_pyr_ELTpup.py             | True   | 0.31665512919425964 |        0 | 0.723645  |
|  7 | test_sh_base.py                | True   | 0.7167708873748779  |        0 | 0.0758622 |
|  8 | test_sh_bpcog.py               | True   | 0.7129252552986145  |        0 | 0.0763685 |
|  9 | test_sh_control_klbasis.py     | True   | 0.7312322854995728  |        0 | 0.075566  |
| 10 | test_sh_cured.py               | True   | 0.8101769089698792  |        0 | 0.210025  |
| 11 | test_sh_ELTpup.py              | True   | 0.5856248140335083  |        0 | 0.0683469 |
| 12 | test_sh_generic_linear.py      | True   | 0.13973799347877502 |        0 | 0.0860589 |
| 13 | test_sh_generic.py             | True   | 0.13973799347877502 |        0 | 0.0750871 |
| 14 | test_sh_geo.py                 | False  | N/A                 |        0 | 0         |
| 15 | test_sh_influBessel.py         | True   | 0.3006800413131714  |        0 | 0.0779258 |
| 16 | test_sh_influBlacknutt.py      | True   | 0.7226215600967407  |        0 | 0.076527  |
| 17 | test_sh_influGaussian.py       | True   | 0.7201576232910156  |        0 | 0.0761665 |
| 18 | test_sh_influRadialSchwartz.py | True   | 0.7247840166091919  |        0 | 0.0792459 |
| 19 | test_sh_influSquareSchwartz.py | True   | 0.7159624695777893  |        0 | 0.0797042 |
| 20 | test_sh_kl.py                  | True   | 0.6966128349304199  |        0 | 0.0665927 |
| 21 | test_sh_lgs_corr.py            | True   | 0.4091269373893738  |        0 | 0.113077  |
| 22 | test_sh_lgs_wcog.py            | True   | 0.6934370398521423  |        0 | 0.107657  |
| 23 | test_sh_modopti.py             | True   | 0.6607018113136292  |        0 | 0.0747513 |
| 24 | test_sh_mv.py                  | True   | 0.7506052255630493  |        0 | 0.0782387 |
| 25 | test_sh_tcog.py                | True   | 0.7167366147041321  |        0 | 0.0790301 |
| 26 | test_sh_wcog.py                | True   | 0.6623543500900269  |        0 | 0.0791775 |
```

The error message seems to be the same for all 4 cases:
```
[1/27] running test_custom_dm_diam_dm_proj
 ** On entry to cusparseCreateCsr() parameter number 5 (csrRowOffsets) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@267 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@275 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@282 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@287 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@295 : invalid value
 ** On entry to cusparseSpMatGetSize() parameter number 1 (spMatDescr) had an illegal value: NULL pointer

 ** On entry to cusparseCreateCsr(): dimension mismatch, nnz (107049168) > matrix size (3600)

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_sparse_obj.cpp@410 : invalid value
 ** On entry to cusparseSpGEMM_copy() parameter number 8 (matC) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense_bufferSize() parameter number 2 (mat_in) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense() parameter number 2 (mat_in) had an illegal value: NULL pointer

Error | carma_csr2dense (sparse) | csr2dense failed
/home/steuer/shesha/tests/check.py:120: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Init"] = str(t_init)
/home/steuer/shesha/tests/check.py:122: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Loop"] = str(t_loop)
```
P.S.: How do you usually run the pytests? They are not run by the checkCOMPASS.py script and the pytest module is not packaged with COMPASS. After installing pytest and running test_compassSupervisor.py I get a couple more failures/warnings:
```
pytest test_compassSupervisor.py 
================================================================================ test session starts =================================================================================
platform linux -- Python 3.11.5, pytest-7.4.0, pluggy-1.0.0
rootdir: /home/steuer/shesha/tests/pytest/supervisor
collected 143 items                                                                                                                                                                  

test_compassSupervisor.py ..........................................s...s.................s.....s.......sss..................s..F..s..F..................................      [100%]

====================================================================================== FAILURES ======================================================================================
_________________________________________________________________________ test_compute_modes_to_volts_basis __________________________________________________________________________

    def test_compute_modes_to_volts_basis():
>       _ = sup.basis.compute_modes_to_volts_basis("KL2V")

test_compassSupervisor.py:552: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../shesha/supervisor/optimizers/modalBasis.py:138: in compute_modes_to_volts_basis
    self.modal_basis = basis.compute_KL2V(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

p_controller = <shesha.config.PCONTROLLER.Param_controller object at 0x7f2e48ba6710>, dms = <sutraWrap.Dms object at 0x7f2ec4e953b0>
p_dms = [<shesha.config.PDMS.Param_dm object at 0x7f2e6b9b4290>, <shesha.config.PDMS.Param_dm object at 0x7f2e48e1bd90>]
p_geom = <shesha.config.PGEOM.Param_geom object at 0x7f2ec4a02310>, p_atmos = <shesha.config.PATMOS.Param_atmos object at 0x7f2e48be7050>
p_tel = <shesha.config.PTEL.Param_tel object at 0x7f2e4a787e90>

    def compute_KL2V(p_controller: conf.Param_controller, dms: Dms, p_dms: list,
                     p_geom: conf.Param_geom, p_atmos: conf.Param_atmos,
                     p_tel: conf.Param_tel):
        """ Compute the Karhunen-Loeve to Volt matrix
        (transfer matrix between the KL space and volt space for a pzt dm)
    
        Args:
    
            p_controller: (Param_controller) : p_controller settings
    
            dms : (shesha_dms) : Dms object
    
            p_dms: (list of Param_dm) : dms settings
    
            p_geom : (Param_geom) : geometry parameters
    
            p_atmos : (Param_atmos) : atmos parameters
    
            p_tel : (Param_tel) : telescope parameters
    
        :return:
    
            KL2V : (np.array(np.float32,dim=2)) : KL to Volt matrix
        """
        ntotact = np.array([p_dms[i]._ntotact for i in range(len(p_dms))], dtype=np.int64)
        KL2V = np.zeros((np.sum(ntotact), np.sum(ntotact)), dtype=np.float32)
    
        indx_act = 0
        nTT = 0
    
        for i in range(p_controller.ndm.size):
            ndm = p_controller.ndm[i]
            if (p_dms[ndm].type == scons.DmType.PZT):
                tmp = (p_geom._ipupil.shape[0] - (p_dms[ndm]._n2 - p_dms[ndm]._n1 + 1)) // 2
                tmp_e0 = p_geom._ipupil.shape[0] - tmp
                tmp_e1 = p_geom._ipupil.shape[1] - tmp
                pup = p_geom._ipupil[tmp:tmp_e0, tmp:tmp_e1]
                indx_valid = np.where(pup.flatten("F") > 0)[0].astype(np.int32)
                p2m = p_tel.diam / p_geom.pupdiam
                norm = -(p2m * p_tel.diam / (2 * p_atmos.r0))**(5. / 3)
    
>               dms.d_dms[ndm].compute_KLbasis(p_dms[ndm]._xpos, p_dms[ndm]._ypos,
                                               indx_valid, indx_valid.size, norm, 1.0)
E               RuntimeError: Caught an unknown exception!

../../../shesha/ao/basis.py:92: RuntimeError
-------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------
Computing KL2V basis...
Computing IF sparse...
-------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------
 ** On entry to cusparseCreateCsr() parameter number 5 (csrRowOffsets) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@267 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@275 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@282 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@287 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_cusparse.cpp@295 : invalid value
 ** On entry to cusparseSpMatGetSize() parameter number 1 (spMatDescr) had an illegal value: NULL pointer

 ** On entry to cusparseCreateCsr() parameter number 5 (csrRowOffsets) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699436963649/work/libcarma/src.cpp/carma_sparse_obj.cpp@410 : invalid value
 ** On entry to cusparseSpGEMM_copy() parameter number 8 (matC) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense_bufferSize() parameter number 2 (mat_in) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense() parameter number 2 (mat_in) had an illegal value: NULL pointer

Error | carma_csr2dense (sparse) | csr2dense failed
_________________________________________________________________________________ test_do_imat_modal _________________________________________________________________________________

    def test_do_imat_modal():
>       sup.calibration.do_imat_modal(0, np.ones(sup.basis.modal_basis.shape[0]), sup.basis.modal_basis)
E       AttributeError: 'NoneType' object has no attribute 'shape'

test_compassSupervisor.py:584: AttributeError
================================================================================== warnings summary ==================================================================================
../../../shesha/util/dm_util.py:359
../../../shesha/util/dm_util.py:359
../../../shesha/util/dm_util.py:359
  /home/steuer/shesha/shesha/util/dm_util.py:359: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
    z[:, :, zn] = z[:, :, zn] + (-1.) ** i * zrmod ** (n - 2. * i) * float(np.math.factorial(n - i)) / \

../../../shesha/util/dm_util.py:360
../../../shesha/util/dm_util.py:360
../../../shesha/util/dm_util.py:360
../../../shesha/util/dm_util.py:360
../../../shesha/util/dm_util.py:360
../../../shesha/util/dm_util.py:360
  /home/steuer/shesha/shesha/util/dm_util.py:360: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
    float(np.math.factorial(i) * np.math.factorial((n + m) // 2 - i) *

../../../shesha/util/dm_util.py:361
../../../shesha/util/dm_util.py:361
../../../shesha/util/dm_util.py:361
  /home/steuer/shesha/shesha/util/dm_util.py:361: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
    np.math.factorial((n - m) // 2 - i))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================== short test summary info ===============================================================================
FAILED test_compassSupervisor.py::test_compute_modes_to_volts_basis - RuntimeError: Caught an unknown exception!
FAILED test_compassSupervisor.py::test_do_imat_modal - AttributeError: 'NoneType' object has no attribute 'shape'
=============================================================== 2 failed, 132 passed, 9 skipped, 12 warnings in 7.74s ================================================================

```

**Comment by a-sevin** [comment]
Hi Horst,

Very weird, I'm using the docker image ```nvidia/cuda:12.3.0-devel-rockylinux8``` to generate binaries and every version of python give the same result at the end (the one for py311)

```md
|    | Test name                      | Init   |   SR@100iter |   T Init |    T Loop |
|---:|:-------------------------------|:-------|-------------:|---------:|----------:|
|  0 | test_custom_dm.py              | True   |     0.702377 |        0 | 0.426822  |
|  1 | test_custom_dm_diam_dm.py      | True   |     0.764012 |        0 | 0.430917  |
|  2 | test_custom_dm_diam_dm_proj.py | True   |     0.744984 |        0 | 0.436008  |
|  3 | test_custom_dm_pitch.py        | True   |     0.76638  |        0 | 0.422093  |
|  4 | test_pyr3_maskedpix.py         | True   |     0.753423 |        0 | 0.131576  |
|  5 | test_pyr_ELTpup.py             | True   |     0.316658 |        0 | 0.138267  |
|  6 | test_pyr_base.py               | True   |     0.642597 |        0 | 0.133989  |
|  7 | test_sh_ELTpup.py              | True   |     0.585377 |        0 | 0.0524276 |
|  8 | test_sh_base.py                | True   |     0.716813 |        0 | 0.0511079 |
|  9 | test_sh_bpcog.py               | True   |     0.712702 |        0 | 0.0726516 |
| 10 | test_sh_control_klbasis.py     | True   |     0.731317 |        0 | 0.0718894 |
| 11 | test_sh_cured.py               | True   |     0.810153 |        0 | 0.080776  |
| 12 | test_sh_generic.py             | True   |     0.139737 |        0 | 0.0552351 |
| 13 | test_sh_generic_linear.py      | True   |     0.139737 |        0 | 0.065052  |
| 14 | test_sh_geo.py                 | True   |     0.850649 |        0 | 0.0546008 |
| 15 | test_sh_influBessel.py         | True   |     0.301636 |        0 | 0.0539837 |
| 16 | test_sh_influBlacknutt.py      | True   |     0.72266  |        0 | 0.0520304 |
| 17 | test_sh_influGaussian.py       | True   |     0.720254 |        0 | 0.059265  |
| 18 | test_sh_influRadialSchwartz.py | True   |     0.724634 |        0 | 0.0529173 |
| 19 | test_sh_influSquareSchwartz.py | True   |     0.715863 |        0 | 0.0515457 |
| 20 | test_sh_kl.py                  | True   |     0.696613 |        0 | 0.0487711 |
| 21 | test_sh_lgs_corr.py            | True   |     0.409171 |        0 | 0.0603399 |
| 22 | test_sh_lgs_wcog.py            | True   |     0.693412 |        0 | 0.0560274 |
| 23 | test_sh_modopti.py             | True   |     0.661344 |        0 | 0.0515385 |
| 24 | test_sh_mv.py                  | True   |     0.750578 |        0 | 0.067141  |
| 25 | test_sh_tcog.py                | True   |     0.716837 |        0 | 0.0519056 |
| 26 | test_sh_wcog.py                | True   |     0.662462 |        0 | 0.056275  |

Resource usage statistics from testing compass:
   Process count: 8
   CPU time: Sys=0:01:27.8, User=0:03:19.6
   Memory: 534.4M
   Disk usage: 109.4K
   Time elapsed: 0:03:02.8


TEST END: /root/miniconda3/conda-bld/linux-64/compass-5.5.0-py3.11_cuda123.rockylinux8.tar.bz2
```

I just finished uploading the ones compiled against CUDA 12.3.0. Could you try this new one?

**Comment by HorstSteuer** [comment]
Same result for the new package:
`compass                   5.5.0           py3.9_cuda123.rockylinux8    compass/label/cuda12`
`NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2 `
```
[1/27] running test_custom_dm_diam_dm_proj
 ** On entry to cusparseCreateCsr() parameter number 5 (csrRowOffsets) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_cusparse.cpp@267 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_cusparse.cpp@275 : invalid value
 ** On entry to cusparseSpGEMM_workEstimation() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_cusparse.cpp@282 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_cusparse.cpp@287 : invalid value
 ** On entry to cusparseSpGEMM_compute() parameter number 8 (matC) had an illegal value: NULL pointer

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_cusparse.cpp@295 : invalid value
 ** On entry to cusparseSpMatGetSize() parameter number 1 (spMatDescr) had an illegal value: NULL pointer

 ** On entry to cusparseCreateCsr(): dimension mismatch, nnz (140286672) > matrix size (3600)

Cusparse error : An unsupported value or parameter was passed to the function.
Cusparse error in /root/miniconda3/conda-bld/compass_1699974357121/work/libcarma/src.cpp/carma_sparse_obj.cpp@410 : invalid value
 ** On entry to cusparseSpGEMM_copy() parameter number 8 (matC) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense_bufferSize() parameter number 2 (mat_in) had an illegal value: NULL pointer

 ** On entry to cusparseSparseToDense() parameter number 2 (mat_in) had an illegal value: NULL pointer

Error | carma_csr2dense (sparse) | csr2dense failed
/home/steuer/shesha/tests/check.py:120: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Init"] = str(t_init)
/home/steuer/shesha/tests/check.py:122: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Loop"] = str(t_loop)

```

**Comment by HorstSteuer** [comment]
Solved! My CUDA version was 12.2. Sorry for the inconvenience! 
After upgrade to 12.3 the tests pass, including the pytest ones (after installing the pytest package), except for a warning:

```
[1/27] running test_custom_dm_diam_dm_proj
/home/steuer/shesha/tests/check.py:120: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Init"] = str(t_init)
/home/steuer/shesha/tests/check.py:122: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0.674976491999999' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[idx, "T Loop"] = str(t_loop)
```
